modules/storage/mysql-single-server/main.bicep

```diff
@@ -0,0 +1,331 @@
+    family: skuFamily
+  }
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {
+    createMode: createMode
+    administratorLogin: administratorLogin
+    administratorLoginPassword: administratorLoginPassword
+    version: version
+    sslEnforcement: sslEnforcement
+    minimalTlsVersion: minimalTlsVersion
+    infrastructureEncryption: infrastructureEncryption
+    storageProfile: {
+      storageMB: storageSizeGB * 1024
+      backupRetentionDays: backupRetentionDays
+      geoRedundantBackup: geoRedundantBackup
+      storageAutogrow: validStorageAutogrow ?? null
+    }
+    publicNetworkAccess: publicNetworkAccess
+    sourceServerId: createMode != 'Default' ? sourceServerResourceId : null
+    restorePointInTime: createMode == 'PointInTimeRestore' ? restorePointInTime : null
+  }
+}
+
+@batchSize(1)
+resource mysqlServerFirewallRules 'Microsoft.DBforMySQL/servers/firewallRules@2017-12-01' = [for firewallRule in firewallRules: {
+  name: firewallRule.name
+  parent: mysqlServer
+  properties: {
+    startIpAddress: firewallRule.startIpAddress
+    endIpAddress: firewallRule.endIpAddress
+  }
+}]
+
+resource mysqlServerVirtualNetworkRules 'Microsoft.DBforMySQL/servers/virtualNetworkRules@2017-12-01' = [for virtualNetworkRule in virtualNetworkRules: {
+  name: virtualNetworkRule.name
+  properties: {
+    ignoreMissingVnetServiceEndpoint: virtualNetworkRule.ignoreMissingVnetServiceEndpoint
+    virtualNetworkSubnetId: virtualNetworkRule.virtualNetworkSubnetId
+  }
+}]
+
+@batchSize(1)
+resource mysqlServerDatabases 'Microsoft.DBforMySQL/servers/databases@2017-12-01' = [for database in databases: {
+  name: database.name
+  parent: mysqlServer
+
+  properties: {
+    charset: database.?charset ?? 'utf32'
+    collation: database.?collation ?? 'utf32_general_ci'
+  }
+}]
+
+@batchSize(1)
+resource mysqlServerConfig 'Microsoft.DBforMySQL/servers/configurations@2017-12-01' = [for configuration in serverConfigurations: {
+  name: configuration.name
+  dependsOn: [
+    mysqlServerFirewallRules
+  ]
+  parent: mysqlServer
+  properties: {
+    value: configuration.value
+    source: 'user-override'
+  }
+}]
+
+@batchSize(1)
+module mysqlRbac 'modules/rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${serverName}-rbac-${index}'
+  params: {
+    description: roleAssignment.?description ?? ''
+    principalIds: roleAssignment.principalIds
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    principalType: roleAssignment.?principalType ?? ''
+    serverName: serverName
+  }
+}]
+
+module mysqlPrivateEndpoint 'modules/privateEndpoint.bicep' = {
+  name: '${serverName}-private-endpoints'
+  params: {
+    location: location
+    privateEndpoints: varPrivateEndpoints
+    tags: tags
+  }
+}
+
+// ------ Diagnostics settings ------
+// May the diagnosticSettings become a module? How can I share the type cross module?
```

Comment:
As above, if I know how to share types between files, I can move this part in a different file